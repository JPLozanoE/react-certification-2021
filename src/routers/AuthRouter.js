/* eslint-disable import/extensions */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { Redirect, Route, Switch } from 'react-router';
import { LoginScreen } from '../components/Auth/LoginScreen';
import { RegisterScreen } from '../components/Auth/RegisterScreen';
import { AuthPage } from '../pages/Auth/Auth.page';

export const AuthRouter = () => {
  return (
    <AuthPage>
      <div className="auth__box-container animate__animated animate__fadeIn animate__faster">
        <Switch>
          <Route path="/auth/login" component={LoginScreen} />
          <Route path="/auth/register" component={RegisterScreen} />
          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </AuthPage>
  );
};
