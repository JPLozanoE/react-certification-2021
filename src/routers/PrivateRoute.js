/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { Redirect, Route } from 'react-router';

export const PrivateRoute = ({ isAuth, component: Component, ...rest }) => {
  // localStorage.setItem('lastPath', rest.location.pathname);
  return (
    <Route
      {...rest}
      component={(props) =>
        isAuth ? <Component {...props} /> : <Redirect to="/auth/login" />
      }
    />
  );
};
