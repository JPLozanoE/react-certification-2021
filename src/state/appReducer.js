import { types } from '../types/types';

export const appReducer = (state = {}, action) => {
  switch (action?.type) {
    case types.changeSearchQuery:
      return {
        ...state,
        search: action.payload,
      };

    case types.setSelectedVideo:
      return {
        ...state,
        selectedVideo: action.payload,
      };

    case types.toggleTheme:
      return {
        ...state,
        isDarkTheme: !state.isDarkTheme,
      };

    case types.setRecommendedVideos:
      return {
        ...state,
        videos: action.payload,
      };

    case types.login:
      return {
        ...state,
        auth: {
          isLogged: true,
          displayName: action.payload.displayName,
          uid: action.payload.uid,
        },
      };

    case types.logout:
      return {
        ...state,
        auth: {
          isLogged: false,
          displayName: '',
          uid: null,
        },
      };

    case types.addFavoriteVideo:
      return {
        ...state,
        favoriteVideos: [
          ...state.favoriteVideos,
          { ...action.payload, isFavorite: true },
        ],
      };

    case types.deleteFavoriteVideo:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: false } : video
        ),
        favoriteVideos: state.favoriteVideos.filter(
          (favoriteVideo) => favoriteVideo.id.videoId !== action.payload
        ),
      };

    case types.updateRecommendedVideos:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: true } : video
        ),
      };

    default:
      return state;
  }
};
