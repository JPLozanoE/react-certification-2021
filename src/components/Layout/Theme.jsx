import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { AppContext } from '../../state/AppContext';

const Theme = ({ children }) => {
  const {
    state: { isDarkTheme },
  } = useContext(AppContext);
  const theme = {
    main: {
      ...(isDarkTheme ? { backgroundColor: '#181818' } : { backgroundColor: '#ffffff' }),
    },
    titles: {
      ...(isDarkTheme ? { color: '#ffffff' } : { color: '#00000' }),
    },
    subtitles: {
      ...(isDarkTheme ? { color: 'rgb(170,170,170)' } : { color: 'rgb(96,96,96)' }),
    },
  };

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default Theme;
