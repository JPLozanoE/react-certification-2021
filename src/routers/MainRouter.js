/* eslint-disable react/jsx-filename-extension */
import React, { useContext, useEffect, useState } from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import GlobalStyles, { FullWidthContainer } from '../__globalStyles';
import { AppContext } from '../state/AppContext';
import { searchYouTube } from '../utils/searchYoutube';
import Layout from '../components/Layout/Layout.component';
import HomePage from '../pages/Home/Home.page';
import { Navbar } from '../components/Home/Navbar';
import { types } from '../types/types';
import { VideoPage } from '../pages/Video/Video.page';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { AuthRouter } from './AuthRouter';
// import { Favorites } from '../components/Favorites/Favorites';
import { FavoritesPage } from '../pages/Favorites/Favorites.page';
import { FavoritesPlayerPage } from '../pages/FavoritesPlayer/FavoritesPlayer.page';

const RedirectImprov = () => <Redirect to="/" />;

function MainRouter() {
  const [error, setError] = useState(false);
  const {
    state: {
      selectedVideo,
      search,
      auth: { isLogged },
    },
    dispatch,
  } = useContext(AppContext);

  console.log(isLogged);

  useEffect(() => {
    const fetchVideos = async () => {
      const res = await searchYouTube(search);
      if (res.error) setError(true);
      dispatch({ type: types.setRecommendedVideos, payload: res.items });
    };
    fetchVideos();
  }, [search, dispatch]);

  return (
    <BrowserRouter>
      <GlobalStyles />
      <Layout>
        <Navbar />
        <FullWidthContainer>
          <Switch>
            <PublicRoute path="/auth" component={AuthRouter} isAuth={isLogged} />
            {error ? (
              <p>Imposible mostrar en este momento</p>
            ) : (
              <>
                <Route exact path="/">
                  <HomePage />
                </Route>
                <Route exact path="/video/:videoId">
                  {selectedVideo !== null ? <VideoPage /> : <Redirect to="/" />}
                </Route>
                <PrivateRoute
                  exact
                  path="/favorites"
                  component={FavoritesPage}
                  isAuth={isLogged}
                />
                <PrivateRoute
                  exact
                  path="/video/favorites/:videoId"
                  component={
                    selectedVideo !== null ? FavoritesPlayerPage : RedirectImprov
                  }
                  isAuth={isLogged}
                />
              </>
            )}
          </Switch>
        </FullWidthContainer>
      </Layout>
    </BrowserRouter>
  );
}

export default MainRouter;
