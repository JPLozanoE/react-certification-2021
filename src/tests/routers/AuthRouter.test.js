/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Theme from '../../components/Layout/Theme';
import { demoState, singleVideo } from '../fixtures/demoState';
import { AppContext } from '../../state/AppContext';
import { AuthRouter } from '../../routers/AuthRouter';

describe('Auth Router', () => {
  let wrapper;

  test('Should render properly when in /auth/register', () => {
    const providerValues = {
      state: { ...demoState, selectedVideo: singleVideo },
      dispatch: jest.fn(),
    };
    const history = createMemoryHistory();
    history.push('/auth/register');
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router history={history}>
            <AuthRouter />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getByRole('heading', { name: 'Register' })).toBeInTheDocument();
  });

  test('Should render properly when in /auth/login', () => {
    const providerValues = {
      state: { ...demoState, selectedVideo: singleVideo },
      dispatch: jest.fn(),
    };
    const history = createMemoryHistory();
    history.push('/auth/login');
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router history={history}>
            <AuthRouter />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(wrapper.getByRole('heading', { name: 'Login' })).toBeInTheDocument();
  });
});
