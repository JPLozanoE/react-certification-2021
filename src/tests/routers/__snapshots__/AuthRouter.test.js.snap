// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Page Should render properly when in /auth/register 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gtsrHT kUvrCA"
      >
        <div
          class="sc-dlnjwi eEaXHH"
        >
          <div
            class="auth__box-container animate__animated animate__fadeIn animate__faster"
          >
            <h3
              class="sc-eCApnc hGciCd auth__title"
            >
              Register
            </h3>
            <form>
              
              <input
                autocomplete="off"
                class="sc-jSFjdj bBOMsQ auth__input"
                name="name"
                placeholder="Name"
                type="text"
                value=""
              />
              <input
                autocomplete="off"
                class="sc-jSFjdj bBOMsQ auth__input"
                name="email"
                placeholder="Email"
                type="text"
                value=""
              />
              <input
                class="sc-jSFjdj bBOMsQ auth__input"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <input
                class="sc-jSFjdj bBOMsQ auth__input"
                name="password2"
                placeholder="Confirm password"
                type="password"
                value=""
              />
              <button
                class="sc-pNWdM kPTJIB btn btn-primary btn-block mb-5"
                type="submit"
              >
                Register
              </button>
              <div>
                <a
                  class="sc-hKFxyN cFwZJz link"
                  href="/auth/login"
                >
                  Already registered?
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtsrHT kUvrCA"
    >
      <div
        class="sc-dlnjwi eEaXHH"
      >
        <div
          class="auth__box-container animate__animated animate__fadeIn animate__faster"
        >
          <h3
            class="sc-eCApnc hGciCd auth__title"
          >
            Register
          </h3>
          <form>
            
            <input
              autocomplete="off"
              class="sc-jSFjdj bBOMsQ auth__input"
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
            <input
              autocomplete="off"
              class="sc-jSFjdj bBOMsQ auth__input"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
            <input
              class="sc-jSFjdj bBOMsQ auth__input"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              class="sc-jSFjdj bBOMsQ auth__input"
              name="password2"
              placeholder="Confirm password"
              type="password"
              value=""
            />
            <button
              class="sc-pNWdM kPTJIB btn btn-primary btn-block mb-5"
              type="submit"
            >
              Register
            </button>
            <div>
              <a
                class="sc-hKFxyN cFwZJz link"
                href="/auth/login"
              >
                Already registered?
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
